src/jwplayer/players/shared/JwPlayerWrapperWithStrategyRules.tsx

```diff
@@ -0,0 +1,116 @@
+			if (viewable === 0 && pauseReason === 'external') {
+				playerInstance.play();
+			}
+		});
+
+		playerInstance.on(JWEvents.PLAY, ({ playReason, viewable }: JWPlayEvent) => {
+			const dismissed = getDismissed();
+			// Pause the content play when the user closed the mini player playing the ad
+			if (dismissed && viewable === 0 && playReason === 'autostart') {
+				playerInstance.pause();
+			}
+		});
+
+		playerInstance.on(JWEvents.READY, () => {
+			recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_READY);
+			triggerVideoMetric('ready');
+			// only add the events after the player is ready
+			jwPlayerPlaybackTracker({ event_name: 'video_player_ready' });
+			addBaseTrackingEvents(playerInstance);
+
+			if (onReady) {
+				console.debug('Player ready...');
+				onReady(playerInstance);
+			}
+		});
+
+		if (typeof onComplete === 'function') {
+			playerInstance.on(JWEvents.COMPLETE, () => onComplete());
+		}
+	};
+
+	const jwPlayerLoaded = (payload: JWPPlacementReadyResponse) => {
+		console.debug('Placement Embed Complete: ', payload.placementId, payload.playerDivId, payload.player);
+
+		// Set the max_resolution param for related videos
+		if (typeof window?.jwplayer?.defaults?.related?.file === 'string') {
+			window.jwplayer.defaults.related.file = window.jwplayer.defaults.related.file + '&max_resolution=1280';
+		}
+
+		jwPlayerPlaybackTracker({ event_name: 'video_player_load' });
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_SCRIPTS_LOAD_READY);
+		triggerVideoMetric('loaded');
+
+		setConfig(config);
+
+		const playerInstance = payload.player;
+		registerEventHandlers(playerInstance);
+		setPlayer(playerInstance);
+	};
+
+	const initPlayer = () => {
+		console.debug('Strategy rules enabled: the embed will be loaded inline', vastUrl);
+	};
+
+	const { setPlayer, setConfig } = useContext(PlayerContext);
+	useBeforeJwpWrapperRendered(initPlayer, shouldLoadSponsoredContentList);
+
+	const selector = `.${parentClassName} .strategyRulesWrapper`;
+	const prerollAdTag = vastUrl ? encodeURIComponent(vastUrl) : '';
+	const strategyRulesUrl = `https://cdn.jwplayer.com/v2/sites/cGlKNUnj/placements/${strategyRulesPlacementId}/embed.js?custom.${strategyRulesPlacementId}.playlist=https://cdn.jwplayer.com/v2/playlists/BdkNc4lb&custom.${strategyRulesPlacementId}.recommendations_playlist_id=${recommendationPlaylistId}&custom.${strategyRulesPlacementId}.preroll_ad_tag=${prerollAdTag}`;
+	const onBeforeLoad = () => {
+		recordVideoEvent(VIDEO_RECORD_EVENTS.JW_PLAYER_SCRIPTS_LOAD_START);
+		jwPlayerPlaybackTracker({ event_name: 'video_player_start_load' });
+	};
+	const onLoad = () => {
+		console.debug('Strategy rules embed loaded. Waiting for player...');
+		window.jwplacements._getPlacementReadyPromise(strategyRulesPlacementId).then(jwPlayerLoaded);
+	};
+
+	useScript(strategyRulesUrl, selector, onBeforeLoad, {
+		onLoad,
+		className,
+		id: jwPlayerContainerEmbedId,
+	});
+
+	return <></>;
```

Comment:
I think it can just return `null`

Reply:
I can try but I think I had it like that before and Typescript was complaining ;)

Reply:
Hm, I think it's OK. Thanks!